### DISCUSSION TITLE
Lua 5.4?

### DISCUSSION THREAD

**Discussion by Codenade** [cc-tweaked/CC-Tweaked#1978] [Q&A] [answered]
Hello everyone,

I'm just curious. Is there any technical limitation that prevents an upgrade to Lua 5.4?  
Of course I understand that an upgrade would probably take a lot of time and effort so I don't expect this to come for some time.

**Comment by SquidDev** [comment, marked as answer]
Most of the limitations are around backwards compatibility. The new integer type causes all sorts of problems (https://github.com/cc-tweaked/CC-Tweaked/issues/950#issuecomment-2116312882), and it's not clear how impactful they are in practice.

Personally, I'm not wild about some of the language changes introduced in 5.3 and 5.4. I understand the motivation, but the execution feels a little clumsy.
